CST-to-AST/generic-functions.lisp

```diff
@@ -20,46 +20,162 @@
 ;;; Generic function CONVERT-FUNCTION-REFERENCE.
 ;;;
 ;;; This generic function converts a reference to a function given by
-;;; a description in the form of an INFO instance.  That INFO instance
-;;; was obtained as a result of a form such as (FUNCTION
-;;; FUNCTION-NAME) by calling FUNCTION-INFO with an environment and
-;;; the FUNCTION-NAME argument.  The function signals an error if the
-;;; INFO instance is not a GLOBAL-FUNCTION-INFO or a
-;;; LOCAL-FUNCTION-INFO.  Client code can override the default
-;;; behavior by adding methods to this function, specialized to the
-;;; particular client defined by that client code.
+;;; a description in the form of an DESCRIPTION instance.
+;;; That DESCRIPTION instance was obtained as a result of a form such as
+;;; (FUNCTION FUNCTION-NAME) by calling DESCRIBE-FUNCTION with
+;;; an environment and the FUNCTION-NAME argument.  The function signals
+;;; an error if the DESCRIPTION instance is not a
+;;; GLOBAL-FUNCTION-DESCRIPTION or a LOCAL-FUNCTION-DESCRIPTION.
+;;; Client code can override the default behavior by adding methods to
+;;; this function, specialized to the particular client defined by that
+;;; client code.
 
-(defgeneric convert-function-reference (client cst info env))
+(defgeneric convert-function-reference (client cst description env))
 
-(defgeneric convert-called-function-reference (client cst info env))
+(defgeneric convert-called-function-reference (client cst description env))
 
 (defgeneric items-from-parameter-group (parameter-group))
 
-(defgeneric convert-global-function-reference (client cst info global-env))
+(defgeneric convert-global-function-reference (client cst description global-env))
 
-(defgeneric convert-special-variable (client cst info global-env))
+(defgeneric convert-special-variable (client cst description global-env))
 
-(defgeneric convert-setq (client var-cst form-cst info env))
+(defgeneric convert-setq (client var-cst form-cst description env))
 
 (defgeneric convert-setq-special-variable
-    (client var-cst form-ast info global-env))
+    (client var-cst form-ast description global-env))
 
 (defgeneric convert-let (client cst environment))
 
 (defgeneric convert-let* (client cst environment))
 
+;; FIXME(paul) Does cst-eval here need two environments or only one?
+;; In the trucler branch of cleavir, it had one only (but two in the
+;; cleavir-environment package).
```

Comment:
I believe that `eval` and `cst-eval` took one environment as the actual environment argument, and one that was just for specialization. This second predates the use of clients, but now that we are using clients we should do that here too. So, probably `cst-eval` should take a form, an environment, and a client as arguments (in some order, i guess the convention is client first?)